### DISCUSSION TITLE
Lite version (View & Router without SSR)

### DISCUSSION THREAD

**Discussion by rrjanbiah** [solidjs/solid#783] [Q&A] [answered]
Apologies if this is already possible, but I can't seem to find it by random searches.

Is there any easy option to get the lite version of the Solid? Especially without the SSR.

Context:

I'm looking to build dynamic SPA site in static hosting such as GitHub pages. As the Solid's view seems to be faster, want to make use of it. I was directed to the starter kit and that launches the server. 

Is there any lite version available already? If not, can you please guide on how to extract that? TIA

**Comment by ryansolid** [comment, marked as answer]
All the basic templates should output something that can be hosted statically for client rendering. You just need to publish the `dist` folder. 

I'd start with either the base JS or TS templates as described here: https://www.solidjs.com/guides/getting-started#getting-started

This creates a new Vite project that is setup for client side rendering. From this it does start a server in dev when you run `npm run dev`. But the `npm run build` command generates static files that can be hosted. I will mention that for Github pages specifically sometimes it takes some hacks to make a SPA work properly, but like Netlify, Vercel, or most other hosts will have an option to redirect all unmatched traffic to index.html which is what you want here. If you using Vercel you want to use the Vite template rather than the Solid Start template.

Hope that helps.

**Reply by rrjanbiah** [reply]
Thank you so much for taking time to reply. Will try and update.

**Reply by rrjanbiah** [reply]
@ryansolid 

> If you using Vercel you want to use the Vite template rather than the Solid Start template.

Would you mind sharing the links of both "Vite template" and "Solid Start template"? I understand that by "template", you're referring to the base starter files in https://github.com/solidjs/templates, but can't understand "Vite template" and "Solid Start template" there. TIA

**Reply by ryansolid** [reply]
I meant in Vercel. They have deployment templates. In terms of project templates any of those should do.

**Reply by rrjanbiah** [reply]
Thanks @ryansolid for being kind here. Will check. Thanks again.
